test-suite/shared/src/test/scala/org/scalajs/testsuite/javalib/lang/CharacterUnicodeBlockTest.scala

```diff
@@ -0,0 +1,1862 @@
+
+          List(
+            s"""assertEquals(UnicodeBlock.$jvmName, UnicodeBlock.forName("$jvmAliasName"))""",
+            s"""assertEquals(UnicodeBlock.$jvmName, UnicodeBlock.forName("$alias"))""",
+            s"""assertEquals(UnicodeBlock.$jvmName, UnicodeBlock.forName("$aliasNoSpaces"))"""
+          )
+        }.getOrElse(Nil)
+      }
+
+      val properNames: List[String] = {
+        List(
+          if (aliases.nonEmpty)
+            s"""assertThrows(classOf[IllegalArgumentException], UnicodeBlock.forName("$jvmFormattedName"))"""
+          else
+            s"""assertEquals(UnicodeBlock.$jvmName, UnicodeBlock.forName("$jvmFormattedName"))""",
+          s"""assertEquals(UnicodeBlock.$jvmName, UnicodeBlock.forName("$properName"))""",
+          s"""assertEquals(UnicodeBlock.$jvmName, UnicodeBlock.forName("$noSpaces"))"""
+        )
+      }
+
+      (properNames ++ aliases).distinct
+    } else {
+      val properNames: List[String] = {
+        List(
+          s"""assertThrows(classOf[IllegalArgumentException], UnicodeBlock.forName("$jvmName"))""",
+          s"""assertThrows(classOf[IllegalArgumentException], UnicodeBlock.forName("$properName"))""",
+          s"""assertThrows(classOf[IllegalArgumentException], UnicodeBlock.forName("$noSpaces"))"""
+        )
+      }
+
+      val aliases: List[String] = {
+        block.jvmAlias.map { alias =>
+          val jvmAliasName = UCDUnicodeBlock.jvmFormattedName(alias)
+          val aliasNoSpaces = alias.replace(" ", "")
+
+          List(
+            s"""assertThrows(classOf[IllegalArgumentException], UnicodeBlock.forName("$jvmAliasName"))""",
+            s"""assertThrows(classOf[IllegalArgumentException], UnicodeBlock.forName("$alias"))""",
+            s"""assertThrows(classOf[IllegalArgumentException], UnicodeBlock.forName("$aliasNoSpaces"))"""
+          )
+        }.getOrElse(Nil)
+      }
+
+      (properNames ++ aliases).distinct
+    }
+  }
+
+  val codePointIntTests: List[String] = {
+    ucdBlocks.flatMap { ucdBlock: UCDUnicodeBlock =>
+
+      makeCodePointTest(
+        ucdBlock,
+        jvmBlockMap.get(ucdBlock.jvmName),
+        useChar = false
+      )
+    }
+  }
+
+  val codePointCharTests: List[String] = {
+    ucdBlocks.flatMap { ucdBlock: UCDUnicodeBlock =>
+      makeCodePointTest(
+        ucdBlock,
+        jvmBlockMap.get(ucdBlock.jvmName),
+        useChar = true
+      )
+    }
+  }
+
+  val nameTests: List[String] = ucdBlocks.flatMap(makeNameTest)
+
+  val generated =
+    s"""
+       |  @Test def of_Char(): Unit = {
+       |    ${codePointCharTests.mkString("\n    ")}
+       |  }
+       |
+       |  @Test def of_CodePoint(): Unit = {
+       |    ${codePointIntTests.mkString("\n    ")}
+       |  }
+       |
+       |  // scalastyle:off line.size.limit
+       |  @Test def forName_String(): Unit = {
+       |    ${nameTests.mkString("\n    ")}
+       |  }
+       |  // scalastyle:on line.size.limit
+       |""".stripMargin
+
+  println(generated)
```

Comment:
Do we really need to test all the cases here? I'm wondering if having a bunch of well chosen test cases wouldn't be enough.

Reply:
*shrug* the exhaustive tests did help isolate issues during development.  I suspect it also helps next time the unicode blocks code gets updated too.  I'm more biased to leave them in, than to incorrectly cherry-pick.  Verifying the JVM character ranges for each unicode block are correct, is also a bonus.

Reply:
I tend to agree with @gzm0 that many of these exhaustive tests seem overkill. I believe it would be more interesting to have well chosen test cases. Personally I would pick the following for the blocks:

* The very first block
* The very last block
* One block in the middle

And the following for the names:

* The ones already in `forName_Normalizations()`, which are very good
* Exhaustive tests for the 3 blocks with a historical name, including the `assertThrows` ones

This set of tests should test all code paths of the implementation, which is I think more interesting than brute-forcing everything.

(Otherwise, LGTM.)

Reply:
PR Updated, for the sake of completeness I have beginning/middle/end of both the Char range and the 2nd surrogate char range